### ISSUE TITLE
Cloud Anchor Hosting Error: ERROR_NOT_AUTHORIZED

### ISSUE THREAD

**Issue by syntax-error404** [google-ar/arcore-android-sdk#681]
Followed the AR Cloud Anchor quickstart guide.  Facing the same issue as [#377](https://github.com/google-ar/arcore-android-sdk/issues/377). **Made sure to put the meta data contating the API key from from Google Cloud service  within the `<application>` tags.**  Set the rules of the database to read and write public. 

Logcat: 
```
2018-12-25 01:39:58.831 18812-19063/com.google.ar.core.examples.java.cloudanchor E/ARCore-AnchorServiceClient: AnchorServiceClient Exception
    blc: PERMISSION_DENIED: The request is missing a valid API key.
        at byp.a(PG:58)
        at byp.a(PG:29)
        at com.google.ar.persistence.AnchorServiceClient.a(PG:16)
        at com.google.ar.persistence.AnchorServiceClient.createAnchors(PG:25)
```

AndoirdManifest:
```
<application
      android:allowBackup="false"
      android:icon="@drawable/ic_launcher"
      android:label="@string/app_name"
      android:theme="@style/AppTheme"
      android:usesCleartextTraffic="false"
      tools:ignore="GoogleAppIndexingWarning">
    <meta-data android:name="com.google.ar.API_KEY"  android:value="MY KEY" />
    <activity
...

**Comment by claywilkinson** [comment]
This most likely means the key not correctly associated with the the application on the API console.  There is more information on how to configure things on the console at [Using API Keys](https://cloud.google.com/docs/authentication/api-keys?ref_topic=6262490&visit_id=636814463557641251-3534710908&rd=1).  Make sure you selected the application you are working with  at the top of the console page when [Enabling Cloud Anchors](https://console.cloud.google.com/apis/library/arcorecloudanchor.googleapis.com)

**Comment by syntax-error404** [comment]
The API Key I'm using says it's compatible with the service. 
![capture1](https://user-images.githubusercontent.com/25047816/50455015-49e3a980-0919-11e9-9e83-356dfa37011a.PNG)

**Comment by claywilkinson** [comment]
Did you create a an OAuth 2.0 client Id for your android application?  If I remember correctly, that is needed in order to look up the API key from the AndroidManifest.xml

**Comment by syntax-error404** [comment]
I have a OAuth 2.0 client Id. Do I also need a service account key?

**Comment by syntax-error404** [comment]
I have done the same but still facing the not authorized error. Is there anything else that needs to be done after creating the OAuth Id?

**Comment by claywilkinson** [comment]
I just tried without using the sample from the SDK, and I just needed an API Key.
What I did was:
1. Create a new project on console.cloud.google.com
2. Then clicked API& Services
3. Clicked Enable API & Services
4. Searched for ARCore and enabled ARCore Cloud Anchor API and enabled it.
5. Went to Credentials and created a new API key (I did not restrict it)
6. Clicked the "copy" icon next to the API key value. 
7. Opened my AndroidManifest.xml and
7.1 Made sure I had the   `<uses-permission android:name="android.permission.INTERNET" />` permission.
7.2  ` <uses-permission android:name="android.permission.CAMERA"/>` was there for ARCore
7.3  `<uses-feature android:name="android.hardware.camera.ar" android:required="true"/> for ARCore`
7.4 in the `<application>` element, I added:
7.4.1 `<meta-data android:name="com.google.ar.core" android:value="required" />` for ARCore
7.4.2 `<meta-data android:name="com.google.android.ar.API_KEY"  android:value="AIzaSyCCCC_ZEI5_FAFStC6ipByYyFwx0xx"/>`  for the API key (I changed the value here to protect the actual value.  I pasted it from the clipboard to make sure there were no spaces and no typos..
